src/common/local_file_system.cpp

```diff
@@ -790,23 +791,32 @@ void LocalFileSystem::MoveFile(const string &source, const string &target) {
 }
 
 FileType LocalFileSystem::GetFileType(FileHandle &handle) {
-	auto path = handle.Cast<WindowsFileHandle>().path;
-	// pipes in windows are just files in '\\.\pipe\' folder
-	if (strncmp(path.c_str(), PIPE_PREFIX, strlen(PIPE_PREFIX)) == 0) {
-		return FileType::FILE_TYPE_FIFO;
-	}
-	DWORD attrs = WindowsGetFileAttributes(path.c_str());
-	if (attrs != INVALID_FILE_ATTRIBUTES) {
-		if (attrs & FILE_ATTRIBUTE_DIRECTORY) {
-			return FileType::FILE_TYPE_DIR;
-		} else {
-			return FileType::FILE_TYPE_REGULAR;
-		}
-	}
-	return FileType::FILE_TYPE_INVALID;
+	return TryGetFileType(handle.Cast<WindowsFileHandle>().path);
+}
+
+// Window pipes are of type CHARDEV
+bool LocalFileSystem::IsPipe(const string &filename) {
+	return GetFileTypeImpl(filename) == FileType::FILE_TYPE_CHARDEV;
 }
 #endif
 
+FileType LocalFileSystem::GetFileTypeImpl(const std::string &path) {
+	string error;
+	auto type = TryGetFileType(path, &error);
+	if (!error.empty()) {
+		throw IOException(error);
+	}
```

Comment:
I am not sure I understand this API.
I would prefer either always throwing (forcing the catch on the caller side) OR passing an optional error message but never throwing here (to be done at the call-sites whenever they feel it make sense).

In particular given the TryGetFileType on Windows never set error, it would mean behaviour would be different (and mostly untested) there, and to me it seems a bit strange.

Reply:
++ let me try to take a look at Windows some more. Part of the reason for the disparity was my unfamiliarity with Windows - but I can take a look!